databricks/koalas/frame.py

```diff
@@ -492,6 +492,104 @@ def to_dict(self, orient='dict', into=dict):
+        mode : str
+            Python write mode, default 'w'.
+        encoding : str, optional
+            A string representing the encoding to use in the output file,
+            defaults to 'ascii' on Python 2 and 'utf-8' on Python 3.
+        compression : str, default 'infer'
+            Compression mode among the following possible values: {'infer',
+            'gzip', 'bz2', 'zip', 'xz', None}. If 'infer' and `path_or_buf`
+            is path-like, then detect compression from the following
+            extensions: '.gz', '.bz2', '.zip' or '.xz'. (otherwise no
+            compression).
+               'infer' option added and set to default.
+        quoting : optional constant from csv module
+            Defaults to csv.QUOTE_MINIMAL. If you have set a `float_format`
+            then floats are converted to strings and thus csv.QUOTE_NONNUMERIC
+            will treat them as non-numeric.
+        quotechar : str, default '\"'
+            String of length 1. Character used to quote fields.
+        line_terminator : string, optional
+            The newline character or character sequence to use in the output
+            file. Defaults to `os.linesep`, which depends on the OS in which
+            this method is called ('\n' for linux, '\r\n' for Windows, i.e.).
+        chunksize : int or None
+            Rows to write at a time.
+        tupleize_cols : bool, default False
+            Write MultiIndex columns as a list of tuples (if True) or in
+            the new, expanded format, where each MultiIndex column is a row
+            in the CSV (if False).
+               This argument will be removed and will always write each row
+               of the multi-index as a separate row in the CSV file.
+        date_format : str, default None
+            Format string for datetime objects.
+        doublequote : bool, default True
+            Control quoting of `quotechar` inside a field.
+        escapechar : str, default None
+            String of length 1. Character used to escape `sep` and `quotechar`
+            when appropriate.
+        decimal : str, default '.'
+            Character recognized as decimal separator. E.g. use ',' for
+            European data.
+        Returns
+        -------
+        None or str
+            If path_or_buf is None, returns the resulting csv format as a
+            string. Otherwise returns None.
+        Examples
+        --------
+        >>> df = ks.DataFrame({'name': ['Raphael', 'Donatello'],
+        ...                    'mask': ['red', 'purple'],
+        ...                    'weapon': ['sai', 'bo staff']})
+        >>> df.to_csv(index=False)
+        'name,mask,weapon\nRaphael,red,sai\nDonatello,purple,bo staff\n'
+        """
+        # Make sure locals() call is at the top of the function so we don't capture local variables.
+        args = locals()
+        kdf = self
+
+        if line_terminator is None and pd.__version__ < "0.24":
```

Comment:
maybe just change our signature to have the line_terminator be \n?

Reply:
I'm not sure if all methods of pandas (version less than 0.24 ) are affected or only this one, that's why I kept it outside `validate_arguments_and_invoke_function`

Reply:
If you choose `\n` explicitly as the default value for us, then we wouldn't have any problem right regardless of the pandas version?

Reply:
I tested with Pandas 0.24 and it works fine! Thanks.

Reply:
Putting "\n" as default value breaks the test for Pandas 0.23. The result is different!
I think it's better we keep it like before. What do you think? 
```
588         Examples
589         --------
590         >>> df = ks.DataFrame({'name': ['Raphael', 'Donatello'],
591         ...                    'mask': ['red', 'purple'],
592         ...                    'weapon': ['sai', 'bo staff']})
593         >>> df.to_csv(index=False)
Expected:
    'name,mask,weapon\nRaphael,red,sai\nDonatello,purple,bo staff\n'
Got:
    'mask,name,weapon\nred,Raphael,sai\npurple,Donatello,bo staff\n'
```

Reply:
@rxin now `\n` is the default value of `line_terminator`. 
The previous error is due to python 3.5